File: python-algorithm/leetcode/problem_2016.py
"""2016. Maximum Difference Between Increasing Elements
https://leetcode.com/problems/maximum-difference-between-increasing-elements/
"""
from typing import List


class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        minimum = nums[0]
        ans = -1
        for i in range(1, len(nums)):
            if nums[i] > minimum:
                ans = max(ans, nums[i] - minimum)
            elif nums[i] < minimum:
                minimum = nums[i]
        return ans

File: java-algorithm/src/main/java/com/leetcode/Problem2016.java
package com.leetcode;

/**
 * 2016. Maximum Difference Between Increasing Elements
 * https://leetcode.com/problems/maximum-difference-between-increasing-elements/
 */
public class Problem2016 {
    public int maximumDifference(int[] nums) {
        int ans = -1;
        int min = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] < min) {
                min = nums[i];
            } else if (nums[i] > min) {
                ans = Math.max(ans, nums[i] - min);
            }
        }
        return ans;
    }
}
